packages/scene-composer/src/components/toolbars/common/ItemContainer.tsx

```diff
@@ -0,0 +1,150 @@
+import { isEmpty } from 'lodash';
+import React, { ReactNode, useCallback, useState } from 'react';
+
+import { getGlobalSettings } from '../../../common/GlobalSettings';
+
+import {
+  Icon,
+  SubMenuIconContainer,
+  ToolbarItemContainer,
+  ToolbarItemIcon,
+  ToolbarItemMenu,
+  ToolbarItemText,
+} from './styledComponents';
+import { ToolbarItemOptions, ToolbarItemOrientation, ToolbarItemType } from './types';
+
+interface ToolbarItemContainerProps {
+  children?: ReactNode;
+  onPointerDown?: React.PointerEventHandler<HTMLDivElement>;
+  onPointerEnter?: React.PointerEventHandler<HTMLDivElement>;
+  onPointerLeave?: React.PointerEventHandler<HTMLDivElement>;
+  onPointerUp?: React.PointerEventHandler<HTMLDivElement>;
+  orientation?: ToolbarItemOrientation;
+  item: ToolbarItemOptions;
+  onItemClick?: (item: ToolbarItemOptions) => void;
+  type: ToolbarItemType;
+  disableSelectedStyle?: boolean;
+}
+
+// Solves forwardRef with props and eslint issue. Explictly setting the props type (twice) silences eslint
+// error "'XXX' is missing in props validation." See https://stackoverflow.com/a/66948051
+export const ItemContainer = React.forwardRef<HTMLDivElement, ToolbarItemContainerProps>(
```

Comment:
What this ItemContainer represents?  it is hard to relate this component on the UI.

Reply:
that's the container for each menu item.